src/components/FlashNotice/FlashNoticeIcon.tsx

```diff
@@ -0,0 +1,24 @@
+import Image from 'next/image';
+
+import { FlashType } from '.';
+
+const FlashNoticeIcon = ({ type }: { type: FlashType }) => {
+  switch (type) {
+    case 'success':
+      return (
+        <Image src="/icon/bell.svg" width={24} height={24} alt="success icon" />
+      );
+
+    case 'warning':
+      return (
+        <Image
+          src="/icon/warning.svg"
+          width={24}
+          height={24}
+          alt="warning icon"
+        />
+      );
+  }
+};
```

Comment:
What about a separation between the logic and the rendering? :) 

```suggestion
  const iconPath = '', iconAlt = '';
  
  switch (type) {
    case 'success':
      iconAlt = 'success icon';
      iconPath = '/icon/bell.svg';

    case 'warning':
      iconAlt = 'warning icon';
      iconPath = '/icon/warning.svg';
  }
  
  return (
    <Image src={iconPath} width={24} height={24} alt={iconAlt} />
  );
};
```

Moving in that direction, we can also leverage the fully-type advantage from TypeScript:

Somewhere before in the file:
```typescript
  type IconOptions = { iconPath: string; iconAlt: string };

  const iconOptions: Record<FlashType, IconOptions> = {
    success: {
      iconAlt: 'success icon',
      iconPath: '/icon/bell.svg',
    },
    warning: {
      iconAlt: 'warning icon',
      iconPath: '/icon/warning.svg',
    },
  };
```

And then we don't need the switch case (we replaced the switch case by a declarative object):

```typescript  
  const options = iconOptions[type];

  return (
    <Image src={options.iconPath} width={24} height={24} alt={options.iconAlt} />
  );
};
```


The final result is 2 lines longer than the original code, but it has the following advantages:
- retrieving the options can easily be replaced by another system later (e.g. adding a prop to the component, geting advanced options from a hook, etc...)
- code is DRYer 
- adding new types does not impact the UI
- the options of the icon are fully typed — i.e. early warnings and auto-completion :) 


**Note:** As your current code has only 2 types, implementing this suggestion is definitely NOT needed. ;-)
(I just wanted to give it a try — for me to learn also ^^)

Reply:
@malparty Nice suggestion. I updated in [6fd0cdf](https://github.com/hanam1ni/react-surveyor/pull/54/commits/6fd0cdfadcce14ceea310c6860469cee70ee7e87), removed switch case and refactored the icon option make the component look much cleaner and organized ✨

Also, TIL about `Record` utility type which very handy. Thank you 👍

Reply:
My pleasure ;-) Happy you like the suggestion :)